### PR TITLE
docs: add markdown version of changelog

### PR DESCRIPTION
**Pull request by tswast**
Changelog converted with

```python
import re


with open("docs/source/changelog.rst") as f:
    text = f.read()

# :issue:`312` -> https://github.com/googleapis/python-bigquery-pandas/issues/312
c = re.compile(r":issue:`([0-9]+)`", flags=re.MULTILINE)

with open("docs/source/changelog.rst", "w", encoding="utf-8") as f:
    f.write(re.sub(c, r"`#\1 <https://github.com/googleapis/python-bigquery-pandas/issues/\1>`_", text))
```

and

```
pandoc --from=rst --to=gfm docs/source/changelog.rst -o CHANGELOG.md
```

Towards https://github.com/googleapis/python-bigquery-pandas/issues/353 and https://github.com/googleapis/python-bigquery-pandas/issues/367

### CODE REVIEW

**Comment by tswast** [comment]
Docs build locally:

```
(dev-3.9) ➜  python-bigquery-pandas git:(release-v0.16.0) nox -s docs
nox > Running session docs
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/docs
nox > python -m pip install -r docs/requirements-docs.txt
nox > python -m pip install -e .
nox > sphinx-build -W -T -N -b html -d docs/source/_build/doctrees/ docs/source/ docs/source/_build/html/
Running Sphinx v4.1.2
making output directory... done
[autosummary] generating autosummary for: api.rst, changelog.md, contributing.rst, howto/authentication.rst, index.rst, install.rst, intro.rst, privacy.rst, reading.rst, writing.rst
loading intersphinx inventory from https://docs.python.org/objects.inv...
loading intersphinx inventory from https://pandas.pydata.org/pandas-docs/stable/objects.inv...
loading intersphinx inventory from https://pydata-google-auth.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://google-auth.readthedocs.io/en/latest/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 10 source files that are out of date
updating environment: [new config] 10 added, 0 changed, 0 removed
reading sources... [ 10%] api
reading sources... [ 20%] changelog
/Users/swast/src/python-bigquery-pandas/.nox/docs/lib/python3.8/site-packages/recommonmark/parser.py:75: UserWarning: Container node skipped: type=document
  warn("Container node skipped: type={0}".format(mdnode.t))
reading sources... [ 30%] contributing
reading sources... [ 40%] howto/authentication
reading sources... [ 50%] index
reading sources... [ 60%] install
reading sources... [ 70%] intro
reading sources... [ 80%] privacy
reading sources... [ 90%] reading
reading sources... [100%] writing

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 10%] api
writing output... [ 20%] changelog
writing output... [ 30%] contributing
writing output... [ 40%] howto/authentication
writing output... [ 50%] index
writing output... [ 60%] install
writing output... [ 70%] intro
writing output... [ 80%] privacy
writing output... [ 90%] reading
writing output... [100%] writing

generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in docs/source/_build/html.
nox > Session docs was successful.
```

File: convert_changelog.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import re
+
+
+with open("docs/source/changelog.rst") as f:
+    text = f.read()
+
+# :issue:`312` -> https://github.com/googleapis/python-bigquery-pandas/issues/312
+c = re.compile(r":issue:`([0-9]+)`", flags=re.MULTILINE)
+print(re.sub(c, r"[#\1](https://github.com/googleapis/python-bigquery-pandas/issues/\1)", text))
```
Comment (jimfulton) [inline convert_changelog.py:9]: Why is this checked in? Isn't it a one-time script?

**Review by jimfulton — COMMENTED** [review]
I can't speak to the automation that this is supposed to interoperate with, but otherwise, this looks reasonable to me.

File: convert_changelog.py (hunk 1 above)
Comment (tswast) [reply convert_changelog.py]: Oops, good catch. Yeah, just a one-time script. Not even the final version I used (which is in the PR description)

**Review by jimfulton — APPROVED** [review]
Modulo automation

**Pull request merged**